python/pydantic_core/core_schema.py

```diff
@@ -163,13 +163,16 @@ class FieldSerializationInfo(SerializationInfo, Protocol):
     def field_name(self) -> str: ...
 
 
-class ValidationInfo(Protocol):
+ContextT = TypeVar('ContextT', covariant=True, default='Any | None')
+
+
+class ValidationInfo(Protocol[ContextT]):
```

Comment:
Should we add some mypy tests for this / maybe also the covariance?

Reply:
We can do so in Pydantic after updating the core version. The covariance is required because it is a `Protocol`, but I'm wondering if it really makes sense. At runtime, it is defined as a pyo3 class: shouldn't we move it's definition to the stub file instead?

Reply:
I guess maybe it is defined this way so that it cannot be instantiated?

Reply:
Ah actually it's not exported by the library, so having it in `_pydantic_core.pyi` seems slightly incorrect, it's not actually importable from there at runtime.

I guess this is why it's defined as a protocol here, the Rust type then satisfies the protocol without needing to actually be related to it in the hierarchy.